When running with MySQL, do not test times with microsecond precision

<b>Comment by <a href='https://openhatch.org/people/paulproteus/'>paulproteus</a>:</b><dl><dd>Right now, it seems, when we run the tests with MySQL, we find test data 
mismatches due to (what seems to be) microsecond precision on times.

You can see this at <a href="https://travis-ci.org/openhatch/oh-mainline/jobs/31520619">https://travis-ci.org/openhatch/oh-mainline/jobs/31520619</a>  -
- search for:

FAIL: test_set_mission_completed_records_first_completed_time 
(mysite.missions.tests.MissionCompletionTestCase)

and you'll find:

Traceback (most recent call last):
  File "/home/travis/build/openhatch/oh-mainline/mysite/missions/tests.py", line 
77, in test_set_mission_completed_records_first_completed_time
    self.assertGreater(obj_after.created_date, now)
AssertionError: datetime.datetime(2014, 8, 2, 19, 26, 33) not greater than 
datetime.datetime(2014, 8, 2, 19, 26, 33, 619950)

We don't care about the 619950 value, which seems to be a microseconds time 
value.

We also might want to change this test to use a mocked-out datetime.datetime, 
but that doesn't seem super necessary.

</dl></dd><hr/>

Status: <b>unread</b>
Nosy List: <b>  paulproteus </b>
Priority: <b>bug</b>
Imported from roundup ID: <b>1026</b> (<a href='http://roundup-archive.openhatch.org/bugs/issue1026'>view archived page</a>)
Last modified: <b>2014-08-04.02:00:10</b>

--- a/mysite/missions/tests.py
+++ b/mysite/missions/tests.py
@@ -24,6 +24,8 @@ import datetime
 
 
 import django.test
+from django.utils.unittest import skipIf
+import django.db
 from mysite.missions.base.view_helpers import *
 from mysite.missions.models import Step, StepCompletion
 from mysite.profile.models import Person
@@ -63,6 +65,9 @@ class MissionCompletionTestCase(django.test.TestCase):
         self.assertTrue(mission_completed(profile, step.name))
         self.assertTrue(obj_after.is_currently_completed)
 
+    @skipIf(django.db.connection.vendor != 'sqlite',
+            'Skipping because only sqlite supports microsecond precision '
+            'with Django. At least, MySQL definitely does not.')
     def test_set_mission_completed_records_first_completed_time(self):
         now = datetime.datetime.now()
         profile = Person.objects.all()[0]

